crypto/fipsmodule/rand/snapsafe_detect_test.cc

```diff
@@ -0,0 +1,121 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/base.h>
+#include <cstdint>
+
+#include <gtest/gtest.h>
+#include "snapsafe_detect.h"
+
+#if defined(OPENSSL_LINUX) && defined(AWSLC_SNAPSAFE_TESTING)
+#include <fcntl.h>
+#include <cstring>
+#include <sys/mman.h>
+
+#define NUMBER_OF_TEST_VALUES 5
+
+typedef struct sgn_test_s {
+  void* addr;
+  size_t pgsize;
+} sgn_test_s;
+
+int init_sgn_file(sgn_test_s* sgn_test);
+int init_sgn_file(sgn_test_s* sgn_test) {
+  const char* sgc_file_path = AWSLC_SYSGENID_PATH;
+  int fd_sgn = open(sgc_file_path, O_CREAT | O_RDWR | O_APPEND, (mode_t)0600);
```

Comment:
What is 0600? Cna that use any of the other flag values to make it clearer?

Reply:
I can change it to [`S_IRUSR | S_IWUSR`](https://en.cppreference.com/w/cpp/filesystem/perms) instead.